superset/databases/utils.py

```diff
@@ -101,3 +103,7 @@ def get_table_metadata(
         "indexes": keys,
         "comment": table_comment,
     }
+
+
+def encode_parameters(parameters: Dict[str, Any]) -> Dict[str, Any]:
+    return {k: urllib.parse.quote(v) for k, v in parameters.items()}
```

Comment:
Let's use `quote_plus` here, since it's what SQLAlchemy recommends: https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=quote_plus#database-urls